src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1413,7 +1416,135 @@ private IBlockOperation CreateBoundBlockOperation(BoundBlock boundBlock)
             ITypeSymbol type = null;
             Optional<object> constantValue = default(Optional<object>);
             bool isImplicit = boundBlock.WasCompilerGenerated;
-            return new CSharpLazyBlockOperation(this, boundBlock, locals, _semanticModel, syntax, type, constantValue, isImplicit);
+            Symbol member = (_semanticModel as MemberSemanticModel)?.MemberSymbol;
+
+            ImmutableArray<ConditionalOperation> prependedNullChecks = ImmutableArray<ConditionalOperation>.Empty;
+            if (!(member is null || member is FieldSymbol))
```

Comment:
>member is null || member is FieldSymbol [](start = 18, length = 39)

These two cases can be included in the `switch`, and then `GeneratedNullChecksForParameters` could be called only if `!parameters.IsEmpty`.

Reply:
I was wondering whether to include them. Thanks for the clarification.